grails-app/controllers/org/bbop/apollo/GroupController.groovy

```diff
@@ -403,51 +387,26 @@ class GroupController {
             , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
             , @RestApiParam(name = "groupId", type = "long", paramType = RestApiParamType.QUERY, description = "Group ID to alter membership of")
             , @RestApiParam(name = "users", type = "JSONArray", paramType = RestApiParamType.QUERY, description = "A JSON array of strings of emails of users the now belong to the group")
+            , @RestApiParam(name = "memberships", type = "JSONArray", paramType = RestApiParamType.QUERY, description = "Bulk memberships (instead of users and groupId) to update of the form: [ {groupId: <groupId>,users: [\"user1\", \"user2\", \"user3\"]}, {groupId:<another-groupId>, users: [\"user2\", \"user8\"]}")
     ]
     )
     @Transactional
     def updateMembership() {
         JSONObject dataObject = permissionService.handleInput(request, params)
-        UserGroup groupInstance = UserGroup.findById(dataObject.groupId)
-        // to support webservice, get current user from session or input object
+
         def currentUser = permissionService.getCurrentUser(dataObject)
-        String creatorMetaData = groupInstance.getMetaData(FeatureStringEnum.CREATOR.value)
-        // allow global admin, group creator, and group admin to update the group membership
-        if (!permissionService.hasGlobalPermissions(dataObject, GlobalPermissionEnum.ADMIN) && !(creatorMetaData && currentUser.id.toString() == creatorMetaData) && !permissionService.isGroupAdmin(groupInstance, currentUser)) {
 
-            render status: HttpStatus.UNAUTHORIZED.value()
-            return
-        }
-        log.info "Trying to update user group membership"
+        if(dataObject.memberships) {
 
+            JSONObject memberships = dataObject.memberships
```

Comment:
Tested! Group creation works fine.
For membership, I have this exception on this line:
```
Cannot cast object '[{"groupId":"1624","users":["..."]}, ...] with class 'org.codehaus.groovy.grails.web.json.JSONArray' to class 'org.codehaus.groovy.grails.web.json.JSONObject' due to: groovy.lang.GroovyRuntimeException: Could not find matching constructor for: org.codehaus.groovy.grails.web.json.JSONObject(org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject, org.codehaus.groovy.grails.web.json.JSONObject)	at org.bbop.apollo.GroupController.$tt__updateMembership(GroupController.groovy:401)
 	at grails.plugin.cache.web.filter.PageFragmentCachingFilter.doFilter(PageFragmentCachingFilter.java:198)
	at grails.plugin.cache.web.filter.AbstractFilter.doFilter(AbstractFilter.java:63)
 	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
         at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
         at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
         at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
         at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
         at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
         at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
         at java.lang.Thread.run(Thread.java:748)
```